spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageProtocolResolverBeanFactoryPostProcessor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.storage;
+
+import com.google.cloud.storage.Storage;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.context.ResourceLoaderAware;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.core.io.ResourceLoader;
+
+/**
+ * A {@link BeanFactoryPostProcessor} implementation to register
+ * {@link GoogleStorageProtocolResolver} with the {@link DefaultResourceLoader}. The
+ * registration is performed only if {@link Storage} bean is declared in the application
+ * context.
+ * <p>
+ * This class must be used as a {@code static @Bean} method definition or just
+ * {@code @Import}ed on some {@code @Configuration} class.
+ * 
```

Comment:
Again checkstyle trailing white space issue. Maybe we should adjust the rule not to apply to comments.